examples/hello_zig/Makefile

```diff
@@ -22,7 +22,7 @@ include $(APPDIR)/Make.defs
 
 # Hello, Zig! Example
 
-MAINSRC = hello_zig_main.zig
+MAINSRC = $(if $(CONFIG_BUILD_KERNEL), hello_zig.zig, hello_zig_main.zig)
```

Comment:
Kernel mode apps expect "main()" function, not sure why the existing source doesn't have it.  The C hello app has "main()" defined and  I remembered for non-KERNEL builds preprocessor macro redefines it to avoid collisions. 

Please teach if there are better ways.

Reply:
Does zig app require a different entry point(e.g. xxx_main) in flat build, but the same entry point(main) in kernel build?

Reply:
Yes I think all apps have such requirements. For example, the sources of "nsh" and "ostest" tell that both entrances are "main()", but if we check the "System.map" for flat build, we can see "nsh_main",  "ostest_main" etc.  This seems not only for Zig.

Reply:
Here is the key point, MAINSRC in flat build will add -Dmain=xxx_main here:
https://github.com/apache/nuttx-apps/pull/2260/files#diff-a095e492c7e42825cec90da21edd36464ab9c3131c2fea9ff970e49a54712af5L259
But, this macro is removed when building as module:
https://github.com/apache/nuttx-apps/pull/2260/files#diff-a095e492c7e42825cec90da21edd36464ab9c3131c2fea9ff970e49a54712af5R243
You can use the same approach as zig.

Reply:
@xiaoxiang781216 I know that. The point is Zig compiler seems don't respect the`-D main=nsh_main`  option. 

If there is way to let Zig compiler accept such options, please teach.